azurelinuxagent/common/protocol/wire.py

```diff
@@ -542,6 +541,8 @@ def __init__(self, endpoint):
         self._endpoint = endpoint
         self._goal_state = None
         self._last_try_update_goal_state_failed = False
+        self._previous_goal_state_timestamp = None
```

Comment:
the timestamp should be in the goal state itself

we want to detach the goal state from the protocol (wire client) so we should not add more dependencies on the properties of the goal state here.

Reply:
Ah, good point, let me take that approach instead.